cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java

```diff
@@ -39,34 +42,32 @@
  */
 public class RESTClient {
 
+  private final List<Listener> listeners;
   private final HttpRequestConfig defaultConfig;
   private final HttpRequestConfig uploadConfig;
   private final int unavailableRetryLimit;
 
-  public RESTClient(HttpRequestConfig defaultConfig, HttpRequestConfig uploadConfig, int retryLimit) {
+  @Inject
+  public RESTClient(ClientConfig clientConfig) {
+    this.listeners = Lists.newArrayList();
```

Comment:
better call this(clientConfig.getDefaultHttpConfig(), clientConfig.getUploadHttpConfig(), clientConfig.getUnavailableRetryLimit());